### PR TITLE
MassInstallAction: Install the CI workflow on this repository

### PR DESCRIPTION
**Pull request by timholy**
This pull request sets up the CI workflow on this repository.

### CODE REVIEW

**Comment by oscardssmith** [comment]
@timholy we should probably update and merge this, right?

**Comment by fingolfin** [comment]
This definitely should be updated and merged. Also the badge in `README.md` should be updated.

Let me know if I can help.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '0.7'
+          - '1.0'
+          - '1.1'
+          - '1.2'
+          - '1.3'
+          - '1.4'
```
Comment (fingolfin) [inline .github/workflows/CI.yml:21]:
```suggestion
          - '1.6'
          - '1.7'
          - '1.8'
          - '1.9'
```

Comment (fingolfin) [reply .github/workflows/CI.yml:21]: Alternatively just test on the first and last supported version to save time
```suggestion
          - '1.6'
          - '1'
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#          - 'nightly'
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
```
Comment (fingolfin) [inline .github/workflows/CI.yml:26]: macOS and Windows runners are more limited in performance and availability. I am not sure they are all that useful for a pure Julia package like Primes.jl... so maybe skip them, or only run them for, say Julia 1 (latest Julia release)
```suggestion
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          path: ~/.julia/artifacts
+          key: ${{ runner.os }}-test-${{ env.cache-name }}-${{ hashFiles('**/Project.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v1
```
Comment (fingolfin) [inline .github/workflows/CI.yml:48]:
```suggestion
      - uses: actions/checkout@v3
      - uses: julia-actions/setup-julia@v1
        with:
          version: ${{ matrix.version }}
          arch: ${{ matrix.arch }}
      - uses: julia-actions/cache@v1
      - uses: julia-actions/julia-buildpkg@v1
      - uses: julia-actions/julia-runtest@v1
      - uses: julia-actions/julia-processcoverage@v1
      - uses: codecov/codecov-action@v3
```

**Pull request closed without merging**